src/Datepicker/Datepicker.tsx

```diff
@@ -22,7 +19,18 @@ import { Icon } from '../Icon'
 import { theme } from '../theme'
 
 import { DatesList } from './DatesList'
-import { Month } from './types'
+
+const getAvailableMonths = (startDate: Date, endDate: Date) => {
+  const monthList = eachMonthOfInterval({
+    start: startDate,
+    end: endDate,
+  })
+
+  return monthList.map((monthDate) => ({
+    date: monthDate,
+    label: format(monthDate, 'MMMM'),
+  }))
+}
```

Comment:
Only the months are generated in advance, although there is probably a case to be made for this to be generated progressively as well
For case where there is no range for example, although it doesn't seems it's possible with the component today